### ISSUE TITLE
Readme file for Samples directory

### ISSUE THREAD

**Issue by gamagan** [google/filament#2287]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
It would be useful to add a Readme file, in the Samples directory, which lists the samples in an order which devs think is a good order to learn Filament. Additionaly, a line about what each sample is about would also be helpful.

**Describe alternatives you've considered**
The other alternative is to look at all the code and try to find an some order to go about the samples.

**OS and backend**
Not OS specific.

**Comment by romainguy** [comment]
What we really need to do is rewrite the samples. They're really more test apps for development than true samples. In particular the foundation they rely on (`samples/app`) is quite messy.

**Comment by prideout** [comment]
Agreed, our Desktop samples are in bad shape.  I wrote a small doc that draws out a plan for this, but implementing the plan has fallen by the wayside.

**Comment by gamagan** [comment]
I'll work with what's currently there. When you guys get a change to do some rewriting, I'll update my knowledge/code based on the new samples. I appreciate the response.
